plugins/eni/engine/dhclient.go

```diff
@@ -185,25 +185,25 @@ func (client *dhclient) constructDHClientArgs(deviceName string, ipRev int) []st
 	}
 	return append(commonArgs,
 		"-lf",
-		client.getLeasesFile(deviceName, ipRev), // The path to the leases file
+		client.getLeasesFile(macAddress, ipRev), // The path to the leases file
```

Comment:
what is the rationale for changing this from device name to mac address?

Reply:
In Kubernetes, CNI passed in eth0 for the CNI_IFNAME. The device is always renamed to eth0 for all Pods and we cannot create the lease file with name in the container network namespace. We also cannot use the host interface name. This is because the name is changing. The name depends on how many network interfaces in host network namespace at the moment. One interface can be eth1 on the host, the be renamed to eth0 in container network namespace and then be changed to eth2 if eth0 and eth1 are there when the pod is destroyed. Therefore, it is preferrable to have a unique identity for each network interface. MacAddress is in use with this context.

Reply:
Thanks. My question was more to the point that today, the lease file is named something like `ns-eth2-dhclient4.leases`. With this change, it seems like, we'll be using the mac address instead of `eth2` IIUC (`ns-aa:bb:..-dhclient4.leases`). Please correct me if i'm wrong.

Reply:
Yes. You are right

Reply:
Thanks for clarifying. I understand it better now.